### DISCUSSION TITLE
Bug confidence score prediction Yolov5

### DISCUSSION THREAD

**Discussion by CesareDavidePace** [obss/sahi#747] [Q&A] [answered]
Hi,
I noticed that if I set the confidence score to 0.1 on yolov5 when predicting with sliced, the results are different than predicting with 0.001 and then filtering out those with a confidence score above 0.1.

**Comment by fcakyon** [comment, marked as answer]
This is a normal behaviour. Filtering predictions before the postprocess and after the postprocess should yield different results.

**Comment by CesareDavidePace** [comment]
quindi se voglio prendere tutte le predizioni e poi filtrare come posso fare?
Devo per forza eseguire l'inferenza a diversi livelli di confidenza?

**Reply by fcakyon** [reply]
```
so if i want to take all the predictions and then filter how can i do?
Do I have to run the inference at different confidence levels?
```
Try using 
- `postprocess_type='NMS'` 
- `postprocess_match_metric='IOU'`
- `postprocess_match_threshold= similar to your original models nms iou threshold`

This way, you should be able to use a lower model threshold in `sahi`.
